streams/src/main/java/org/apache/kafka/streams/processor/assignment/TopicPartitionAssignmentInfo.java

```diff
@@ -0,0 +1,45 @@
+package org.apache.kafka.streams.processor.assignment;
+
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * This is a simple container class used during the assignment process to distinguish
+ * TopicPartitions type. Since the assignment logic can depend on the type of topic we're
+ * looking at, and the rack information of the partition, this container class should have
+ * everything necessary to make informed task assignment decisions.
+ */
+public interface TopicPartitionAssignmentInfo {
```

Comment:
the only thing I take issue with is the name here -- technically this is just application metadata about how this topic partition fits into the topology, ie it's not really "assignment" info -- we're just using it for assignment purposes

...I'm not sure I'm explaining that well and it sounds overly pedantic, but if I didn't bring it up now then someone probably would say the same thing on the KIP discussion once we send out this update.

I know `TopicPartitionInfo` is already taken, and is too broad anyways. I'd recommend something like `TaskTopicPartition` which I think helps drive home the point that this is metadata related to how this topic partition relates to the given task, not info about the topic partition itself per say

Does that make any sense?

Reply:
To elaborate on what I said I believe that generally speaking interfaces that are too broad when created get super bloated over time, so I usually err on the side of small and precise and expand if necessary.

Reply:
I mean I do agree on keeping it scoped to the assignment, and putting it under the assignment package should be effective enough. I was more thinking along the lines of marking a clear distinction between interfaces/metadata that falls under the `ApplicationState`/input status vs things that fall under the `TaskAssignment`/output category. Unfortunately "assignment" is just an overloaded term, but since it's correlated with assignor output in other class names, I think it's best not to use it in class names to also mean "related to the assignor" since putting it under the assignment package is another way to signal that